File: android/app/src/main/java/com/vegettable/app/model/ApiResponse.java
package com.vegettable.app.model;

import com.google.gson.annotations.SerializedName;

/**
 * .NET API 統一回應格式
 */
public class ApiResponse<T> {
    @SerializedName("success")
    private boolean success;

    @SerializedName("data")
    private T data;

    @SerializedName("message")
    private String message;

    @SerializedName("timestamp")
    private long timestamp;

    public boolean isSuccess() { return success; }
    public T getData() { return data; }
    public String getMessage() { return message; }
    public long getTimestamp() { return timestamp; }
}

File: android/app/src/main/java/com/vegettable/app/model/ApiResponse.kt
package com.vegettable.app.model

import com.google.gson.annotations.SerializedName

/**
 * .NET API 統一回應格式
 */
class ApiResponse<T> {
    @SerializedName("success")
    val isSuccess: Boolean = false

    @JvmField
    @SerializedName("data")
    val data: T? = null

    @SerializedName("message")
    val message: String? = null

    @SerializedName("timestamp")
    val timestamp: Long = 0
}
